src/newgrf_spritegroup.cpp

```diff
@@ -36,17 +36,20 @@ TemporaryStorageArray<int32, 0x110> _temp_store;
 {
 	if (group == nullptr) return nullptr;
 
-	if (_newgrf_profiler == nullptr || _newgrf_profiler->grffile != object.grffile) {
+	const GRFFile *grf = object.grffile;
+	auto profiler = std::find_if(_newgrf_profilers.begin(), _newgrf_profilers.end(), [&](const NewGRFProfiler &pr) { return pr.grffile == grf; });
```

Comment:
This can only change if top_level==true, so it may be worth caching somewhere.

Reply:
I don't think it's worth optimising this, for regular gameplay the profilers list will be empty so in -O2 builds the entire find_if hopefully compiles to have a fast is-empty path, and if the player is operating the profiler they will have to put up with a small bit of extra slowness.